plone/app/content/browser/vocabulary.py

```diff
@@ -1,41 +1,48 @@
 # -*- coding: utf-8 -*-
 from AccessControl import getSecurityManager
-from plone.app.layout.navigation.interfaces import INavigationRoot
-from plone.app.layout.navigation.root import getNavigationRoot
-from Products.Five import BrowserView
 from logging import getLogger
 from plone.app.content.utils import json_dumps
 from plone.app.content.utils import json_loads
+from plone.app.layout.navigation.interfaces import INavigationRoot
+from plone.app.layout.navigation.root import getNavigationRoot
 from plone.app.querystring import queryparser
 from plone.app.widgets.interfaces import IFieldPermissionChecker
 from plone.autoform.interfaces import WRITE_PERMISSIONS_KEY
 from plone.supermodel.utils import mergedTaggedValueDict
+from Products.CMFPlone import PloneMessageFactory as _
+from Products.CMFPlone.utils import safe_unicode
+from Products.Five import BrowserView
 from types import FunctionType
 from zope.component import getUtility
 from zope.component import queryAdapter
 from zope.component import queryUtility
+from zope.deprecation import deprecated
+from zope.i18n import translate
 from zope.schema.interfaces import ICollection
 from zope.schema.interfaces import IVocabularyFactory
 from zope.security.interfaces import IPermission
-from Products.CMFPlone import PloneMessageFactory as _
-from zope.i18n import translate
-from Products.CMFPlone.utils import safe_unicode
+
 import inspect
 import itertools
 
+
 logger = getLogger(__name__)
 
 MAX_BATCH_SIZE = 500  # prevent overloading server
 
-_permissions = {
+DEFAULT_PERMISSION = 'View'
+PERMISSIONS = {
     'plone.app.vocabularies.Users': 'Modify portal content',
     'plone.app.vocabularies.Catalog': 'View',
     'plone.app.vocabularies.Keywords': 'Modify portal content',
     'plone.app.vocabularies.SyndicatableFeedItems': 'Modify portal content'
 }
 
+_permissions = deprecated(PERMISSIONS, 'Use PERMISSION variable instead')
```

Comment:
The `deprecated` call cannot handle a dictionary and returns `None`. So `_permissions` is `None` after this call. I noticed this in ploneintranet, which imported `_permissions` and got into problems because it was `None`.

`zope.deprecated` handle strings, tuples, lists, methods, and functions. Not dictionaries. I tried adding that, but you run into an error `cannot hash dictionary`. The list which is does support, also cannot be just any list, but is meant to be a list of strings which are variable names that have been imported or defined in the current module.
In other words, this is not the way. :-)

Fixed in pull request #95.